syntax = "proto3";

package msk_test;

// 示例消息
message UserEvent {
    int64 user_id = 1;
    string username = 2;
    string action = 3;
    int64 timestamp = 4;
    map<string, string> metadata = 5;
}

// 订单消息示例
message OrderEvent {
    string order_id = 1;
    int64 user_id = 2;
    repeated OrderItem items = 3;
    double total_amount = 4;
    OrderStatus status = 5;
    int64 created_at = 6;
}

message OrderItem {
    string product_id = 1;
    string product_name = 2;
    int32 quantity = 3;
    double price = 4;
}

enum OrderStatus {
    PENDING = 0;
    CONFIRMED = 1;
    SHIPPED = 2;
    DELIVERED = 3;
    CANCELLED = 4;
}

